component/componenttest/nop_host.go

```diff
@@ -29,14 +31,21 @@ func NewNopHost() component.Host {
 	return nopHostInstance
 }
 
+// NewNopHost returns a new instance of nopHost with proper defaults for most tests.
+func NewNopHostWithExtensions(ext map[config.ComponentID]component.Extension) component.Host {
```

Comment:
No longer no-op. This can be easily implemented by embedding the interface and overwrite the GetExtension, don't think it is worth the extra API

Reply:
Good point, embedded the interface in the test, PTAL.